src/github.com/matrix-org/dendrite/clientapi/routing/routing.go

```diff
@@ -185,23 +185,17 @@ func Setup(
 		}),
 	).Methods("GET")
 
-	r0mux.Handle("/user/{userID}/filter",
-		common.MakeExternalAPI("make_filter", func(req *http.Request) util.JSONResponse {
-			// TODO: Persist filter and return filter ID
-			return util.JSONResponse{
-				Code: 200,
-				JSON: struct{}{},
-			}
+	r0mux.Handle("/user/{userId}/filter",
+		common.MakeAuthAPI("put_filter", deviceDB, func(req *http.Request, device *authtypes.Device) util.JSONResponse {
+			vars := mux.Vars(req)
+			return writers.PutFilter(req, device, accountDB, vars["userId"])
 		}),
-	).Methods("POST", "OPTIONS")
```

Comment:
What was this needed for? I haven't found that in the spec.

Reply:
We need the OPTIONS to satisfy CORS request pre-flighting https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request otherwise it's hard to hit the API from web clients.

Reply:
oh, okay, so I'll readd this. Should I go ahead and edit the current commit, or are multiple commits maybe better? The first would make the git history nicer, the latter would make the review and evolution inside this PR look nicer.

Reply:
We squash merge so it doesn't really matter :)

Reply:
okay, nice 👍

Reply:
About the OPTIONS thingy there, I do not need to add more than this additional method, do I? I couldn't find more handling of OPTIONS at some other points, so I guess the go http package just handles that on his own?

Reply:
No, you don't need to do anything more. It gets handled by the wrapper i think